### PR TITLE
Svg Icons

### PR DESCRIPTION
**Pull request by mcraiganthony**
SvgIcon component.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,4 @@ node_modules/
 .orig
 www/
 npm-debug.log
+bower_components/
```
Comment (mfunkie) [inline .gitignore]: I'm cool with keeping this, but it looks like we don't use it anymore?

File: index.html [hunk 2]
```diff
@@ -19,10 +19,20 @@
...
         <script type="text/javascript" src="//cdnjs.cloudflare.com/ajax/libs/codemirror/5.0.0/codemirror.min.js"></script>
-<script type="text/javascript" src="//cdnjs.cloudflare.com/ajax/libs/codemirror/5.0.0/mode/javascript/javascript.min.js"></script>
+        <script type="text/javascript" src="//cdnjs.cloudflare.com/ajax/libs/codemirror/5.0.0/mode/javascript/javascript.min.js"></script>
         <script src="//fb.me/react-with-addons-0.13.1.js"></script>
         <script src="boomstrap-react.js"></script>
         <script src="vendor.bundle.js"></script>
         <script src="boomstrap-react-docs.js"></script>
+
+        <!-- load boomstrap's js to give us access to boomsvgloader -->
+        <script src="//boomtownroi.github.io/boomstrap/js/boomstrap.min.js"></script>
```
Comment (mfunkie) [inline index.html]: Could we just npm install boomsvgloader and load it locally?

File: src/Components/SvgIcon.jsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  render() {
+    let { className, icon, ...props } = this.props;
+
+    let useTag = `<use xlink:href=#icon-${icon} />`;
+
+    const classes = cx(
+      className,
+      'icon',
+      'icon-' + icon
```
Comment (mfunkie) [inline src/Components/SvgIcon.jsx]: If we're using SVG, is this relevant? Or are there still class stuff to be gotten from this.

File: src/Components/SvgIcon.jsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    );
+
+    const createUseTag = () => {
```
Comment (mfunkie) [inline src/Components/SvgIcon.jsx]: You don't need a function here if you're only calling it once.  Maybe just do

```js
const useTagHtml = {
  __html: useTag
};
```

File: .gitignore (hunk 1 above)
Comment (mcraiganthony) [inline .gitignore]: Need to keep it there just in case someone isn't careful down the road.

File: index.html (hunk 2 above)
Comment (mcraiganthony) [inline index.html]: Yeah. That's a good idea.

File: src/Components/SvgIcon.jsx (hunk 3 above)
Comment (mcraiganthony) [inline src/Components/SvgIcon.jsx]: The 'icon' class is absolutely relevant. The 'icon-' class isn't necessary for EVERY icon. But for some of them, we use the class to alter the icon's top position. Sometimes these icons need to jog up or down a tad depending on how they are sitting on the grid. I'd prefer to keep them there.

Comment (mfunkie) [inline src/Components/SvgIcon.jsx]: That's cool.  I just wasn't sure.

File: src/Components/Switcher.jsx [hunk 5]
```diff
@@ -1,5 +1,6 @@
 const React = require('react/addons');
 const cx    = require('classnames');
+const Icon                  = require('./Icon');
```
Comment (mfunkie) [inline src/Components/Switcher.jsx]: Weird spacing here.

**Comment by mcraiganthony** [comment]
@mfunkie Do @igreulich's changes look okay?

**Comment by mfunkie** [comment]
Yup!

**Comment by mfunkie** [comment]
Super fun merge time!

**Pull request merged**